bundled/kokkos-4.5.01/core/src/Kokkos_MemoryPool.hpp

```diff
@@ -458,7 +458,7 @@ class MemoryPool {
   /* Return 0 for invalid block size */
   KOKKOS_INLINE_FUNCTION
   uint32_t allocate_block_size(uint64_t alloc_size) const noexcept {
-    return alloc_size <= (1UL << m_max_block_size_lg2)
+    return alloc_size <= (uint64_t(1) << m_max_block_size_lg2)
```

Comment:
@masterleinad The issue here is as follows: In the existing Kokkos code, on Windows (on which `long unsigned int` is 32-bit), the type of 
```
1UL << m_max_block_size_lg2
```
is 32-bit unsigned int. It is compared against `alloc_size`, which is 64-bit unsigned int. So the compiler promotes the rhs to 64-bit as well. It then warns by saying "well, if you store the result in the rhs as a 64-bit int after all, did you actually mean to do the shift in 64-bit originally?" In this patch, I essentially say "yes, yes, that seems like a good idea".

The issue arises because on all sane (LP64) platforms, `1UL` is a 64-bit data type. On Windows, `long int = int = int32_t`, so `1UL=1U`.

Reply:
I made a small adjustment here and then upstreamed the patch to Kokkos in https://github.com/kokkos/kokkos/pull/7885

Reply:
I'm somewhat surprised that we are only seeng this warning for MSVC. For other compilers, we explicitly ignore warnings via https://github.com/dealii/dealii/blob/fc11082d3eab12e8a0687f1d10331cf6bcd5cbfa/bundled/kokkos-3.7.00/CMakeLists.txt#L66-L69 and I had suggested to add something for MSVC if there'd be more warnings (that are turned into errors).

Reply:
Sure, we could also attack the issue there. I don't know enough about the Windows compiler to say which flag to use. Though I'm generally more in favor of just fixing the problem upstream if we can -- which here we actually can.